src/platform/packages/shared/response-ops/rule_params/es_query/v1.ts

```diff
@@ -200,6 +200,35 @@ function validateParams(anyParams: unknown): string | undefined {
     termSize,
   } = anyParams as EsQueryRuleParams;
 
+  if (isEsqlQueryRule(searchType)) {
+    const { timeField } = anyParams as EsQueryRuleParams;
+
+    if (!timeField) {
+      return i18n.translate('xpack.responseOps.ruleParams.esQuery.esqlTimeFieldErrorMessage', {
+        defaultMessage: '[timeField]: is required',
+      });
+    }
+    if (thresholdComparator !== Comparator.GT) {
+      return i18n.translate(
+        'xpack.responseOps.ruleParams.esQuery.esqlThresholdComparatorErrorMessage',
+        {
+          defaultMessage: '[thresholdComparator]: is required to be greater than',
+        }
+      );
+    }
+    if (threshold && threshold[0] !== 0) {
+      return i18n.translate('xpack.responseOps.ruleParams.esQuery.esqlThresholdErrorMessage', {
+        defaultMessage: '[threshold]: is required to be 0',
+      });
+    }
+    if (groupBy === 'top') {
```

Comment:
Does this check break backwards compatibility for users who may have accidentally added top as a `groupBy` field?

Reply:
I'm not sure I follow the scenario where we'd have to support someone who did use "all"? Using "all" in ES|QL instead of "top" makes sense to me and I don't think we'll need to support "top" down the road given that logic belongs in ES|QL.

Reply:
Yea, I don't know that we can add this validation now I guess just in case people are copying and pasting things and get it wrong in the API request before?

Reply:
That could be one way, and we could fail validation when ES|QL is specified -- unless the validation already allowed this before the PR, then we'd have a breaking change.

Reply:
> unless the validation already allowed this before the PR, then we'd have a breaking change (having APIs starting to reject this)

I believe that was the case.  We weren't validating the field when it was used with ES|QL before, so users could pass any of the allowed values ('top' or 'all', at the time - now we have 'row' with this PR).  So if someone copy/pasta'd a 'top' in there before this PR, it would have acted like 'all' since that was previously the only "mode" for this type.  If we validate it, the rule would then fail.

So I think removing validation is correct.

Feels like perhaps we should add a comment in that validator indicating WHY we aren't validating `groupBy` and mentioning we will be treating 'top` as 'all'.  We'll forget why we did this in a few months :-)

Reply:
Comment added in this commit, [2e39beb](https://github.com/elastic/kibana/pull/212135/commits/2e39beb117c98ef2be06cd04b7828c9d6deac171)